app/controllers/application_controller.rb

```diff
@@ -5,6 +5,176 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
+  end
+
+  get '/' do 
+   
+    erb :index
+  end 
+
+
+  get '/login' do 
+    if !is_logged_in?
+        erb :'users/login'
+    else 
+      redirect '/tweets'
+    end 
+  end 
+
+
+
+  get '/signup' do 
+    
+      if is_logged_in?
+          redirect '/tweets'
+      else 
+      erb :'users/create_user'
+    end 
+  end
+
+  post '/signup' do 
+   
+      if params[:username].present? && params[:email].present? && params[:password].present?
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.